fecgov/fec-cms
# replace nswatch with npm-watch and concurrently

## Summary (required)

- Resolves #5567 

This ticket replaces nswatch with npm-watch and concurrently. 

#### How we use nswatch:  
- We watch three directories for front-end changes 
    - `./fec/fec/static/js/**/*.js`
    - `./fec/fec/static/js/templates/**/*.hbs`
    - `./fec/fec/static/scss/**/*.scss`
- On startup it executes build-js and build-sass
- The watcher continues running in the terminal 
- Upon changes to js. or .hbs files it automatically reruns build-js
- Upon changes to .scss files it automatically reruns build-sass

Concurrently allows npm-watch to re-run the build scripts based on which type of file has changed. Using npm-watch and concurrently together has the same functionality as nswatch.

Documentation: 
[Concurrently](https://www.npmjs.com/package/concurrently)
[npm-watch](https://www.npmjs.com/package/npm-watch)


### Required reviewers 1 - 2 developers


## Impacted areas of the application

General components of the application that this PR will affect:

-  package.json



## How to test
1. Checkout Branch
2. `npm install`
3. `npm run watch`
4. Make changes to any file in ./fec/fec/static/js/ with the extension .js or .hbs, and ensure that build-js re-runs automatically. 
5. Make changes to any file in ./fec/fec/static/scss/ with the extension .scss, and ensure that build-sass reruns automatically.
6. Ctrl-c in termnial to terminate program 
7. `cd fec`
8. `./manage.py runserver`
9. Check that changes are reflected.
